bindings/rust/s2n-tls/src/enums.rs

```diff
@@ -7,6 +7,7 @@ use crate::error::Error;
 use core::convert::TryFrom;
 use s2n_tls_sys::*;
 
+#[allow(clippy::derive_partial_eq_without_eq)]
```

Comment:
Do we have to inline this everywhere, or can we turn it off for entire projects?

Reply:
I didn't go for globally disabling this clippy warning because I don't know how useful the warning is ie. if a new struct is added and does not derive Eq w/ Partial_Eq then will the warning help? If the answer is it is never helpful then I can move it into the `clippy.toml` file

Reply:
From the discussion in https://github.com/aws/s2n-tls/pull/3448#discussion_r946194761, it sounds like it's either never helpful or at least rarely helpful. The bindings are mostly public enums, since there's very little internal logic. Let's add it to clippy.toml.

Reply:
Makes sense. I'll read more about this and add it to `clippy.toml`

Reply:
Actually, it looks like Clippy doesn't support disabling this lint in `clippy.toml`. Apparently, [only a limited set of lints](https://github.com/rust-lang/rust-clippy#configuration) are applicable. I think we're forced to use the in-code disabling

Reply:
you can add `#![allow(clippy::derive_partial_eq_without_eq)]` to `bindings/rust/s2n-tls/src/lib.rs`

https://doc.rust-lang.org/reference/attributes.html
> Inner attributes, written with a bang (!) after the hash (#), apply to the item that the attribute is declared within.

Reply:
Done